### PR TITLE
Fix loop

### PR DESCRIPTION
**Pull request by romkatsu**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #8

### CODE REVIEW

File: src/DataResponse.php [hunk 1]
```diff
@@ -164,4 +164,14 @@ public function getData()
         }
         return is_object($this->data) ? clone $this->data : $this->data;
     }
+
+    private function formatResponse(): ResponseInterface
+    {
+        $response = $this->responseFormatter->format($this);
+        if ($response instanceof self) {
+            throw new \RuntimeException('Loop has been detected. Formatted response cannot be DataResponse.');
```
Comment (samdark) [inline src/DataResponse.php:172]: Is it possible to improve this message so it's clear what exactly is wrong and how to fix it?

Comment (romkatsu) [reply src/DataResponse.php:172]: "DataResponseFormatterInterface should return no DateResponse" ?

Comment (yiiliveext) [reply src/DataResponse.php:172]: DataResponseFormatterInterface should not return instance of DateResponse

Comment (romkatsu) [reply src/DataResponse.php]: rename

**Comment by samdark** [comment]
👍

**Pull request merged**